File: src/main/java/com/devsuatt/todoApp/dto/CreateTaskRequestDto.java
package com.devsuatt.todoApp.dto;

import com.devsuatt.todoApp.model.TaskType;
import lombok.AllArgsConstructor;
import lombok.Getter;

import javax.validation.constraints.NotBlank;
import java.time.LocalDateTime;

@AllArgsConstructor
@Getter
public class CreateTaskRequestDto {
    @NotBlank(message = "userId can not be blank!")
    private String userId;
    @NotBlank(message = "task header can not be blank!")
    private String taskHeader;
    private String taskDescription;

}

File: src/main/java/com/devsuatt/todoApp/dto/CreateTaskRequestDto.kt
package com.devsuatt.todoApp.dto

import com.devsuatt.todoApp.model.TaskType
import java.time.LocalDateTime
import javax.validation.constraints.NotBlank

data class CreateTaskRequestDto(

    @field:NotBlank(message = "userId can not be null!")
    val userId: String,
    @field:NotBlank(message = "task header can not be null!")
    val taskHeader: String,
    val taskDescription: String?,


    )
